recipes/newrelic/infrastructure/ohi/memcached/debian.yml

```diff
@@ -81,49 +66,107 @@ install:
     setup:
       label: "Installing Memcached integration..."
       cmds:
+        - sudo apt-get install libmemcached-tools -y > /dev/null
         - |
           sudo mkdir -p "/etc/newrelic-infra/integrations.d"
         - |
           # Get latest definitions and skip any failure because of deprecation
           sudo apt-get -o Acquire::Check-Valid-Until=false update -yq
         - |
-          sudo apt-get install nri-memcached -y
+          sudo apt-get install nri-memcached -y > /dev/null
         - |
           if [ -f /etc/newrelic-infra/integrations.d/memcached-config.yml ]; then
             sudo rm /etc/newrelic-infra/integrations.d/memcached-config.yml;
           fi
         - |
-          if [ -z {{.NR_CLI_DB_USERNAME}} ] && [ -z {{.NR_CLI_DB_PASSWORD}} ]; then
-            sudo tee /etc/newrelic-infra/integrations.d/memcached-config.yml > /dev/null <<"EOT"
+          
+          # try get values from env
+          NR_CLI_DB_USERNAME={{.NR_CLI_DB_USERNAME}}
+          NR_CLI_DB_PASSWORD={{.NR_CLI_DB_PASSWORD}}
+          NR_CLI_HOST={{.NR_CLI_HOST}}
+          NR_CLI_PORT={{.NR_CLI_PORT}}
+
+          # Set default values for variables
+          NR_CLI_HOST=${NR_CLI_HOST:-localhost}
+          NR_CLI_PORT=${NR_CLI_PORT:-11211}
+          MAX_RETRIES=3
+          TRIES=0 
+          CAN_CONNECT=$(curl $NR_CLI_HOST:$NR_CLI_PORT 2>&1 1>/dev/null -s -S | awk -F'[()]' '{print $2}')
+
+          while [[ $TRIES -lt $MAX_RETRIES  && $CAN_CONNECT == "7" ]]; do
+            printf "\nMemcached Port (default: 11211): "
+            read -r NR_CLI_DB_PORT
+            NR_CLI_DB_PORT=${NR_CLI_DB_PORT:-11211}
+            ((TRIES++))
+
+            CAN_CONNECT=$(curl $NR_CLI_HOST:$NR_CLI_PORT 2>&1 1>/dev/null -s -S | awk -F'[()]' '{print $2}')
+
+            if [ $CAN_CONNECT == "7" ]; then
+              printf "\n\nFail to connect to Memcached on localhost using port 11211.  Please enter the port number for your Memcached instance.\n" >&2
```

Comment:
@NRhzhao sorry, but the "Fail" semantic is a bit harsh to users, again particularly if users just happen to use a different port than the standard port and were never asked to enter a port number.
Maybe we could just trim the beginning and keep the "Please enter the port number for your Memcached instance", since we know the default doesn't work, we don't have to display it and have them find out their own port number without knowing what the default is anyway.

Reply:
updated